app/src/main/java/com/example/sagar/popupshops_buyerside/Shop/add.java

```diff
@@ -111,11 +135,27 @@ public void onFailure(@NonNull Exception e) {
                             }).addOnSuccessListener(new OnSuccessListener<UploadTask.TaskSnapshot>() {
                                                         @Override
                                                         public void onSuccess(UploadTask.TaskSnapshot taskSnapshot) {
-                                                            dbRef.setValue(
+                                                            itemRef.setValue(
                                                                     new Item(
                                                                             categoryString, Float.parseFloat(priceString), description, taskSnapshot.getMetadata().getDownloadUrl().toString(), Integer.parseInt(stockString)
                                                                     )
                                                             );
+
+                                                            categoryRef.runTransaction(new Transaction.Handler() {
```

Comment:
If category already exists, then dont add category to the generalized category table